### PR TITLE
Added support for custom validators

### LINKED ISSUE #85: @schema.validate allow to transfer custom `IValidator` class that will be used to validate the instance.
Hi,

I want use own jsonschema validator. 
Can you accept it in @schema.validate decorator
and than pass it to jsonschema.validate()

Same way as you doing with `format_checker`.

### PR DESCRIPTION
**Pull request by mauler**
Resolves #85

### CODE REVIEW

File: tests/func_test.py [hunk 1]
```diff
@@ -111,6 +144,18 @@ def get_app(self):
             db_conn=None
         )
 
+    def test_post_custom_validator_class(self):
+        """"""
```
Comment (hfaran) [inline tests/func_test.py]: Don't need this if you're not going to put anything in it. May as well remove it.

File: tests/func_test.py [hunk 2]
```diff
@@ -31,6 +33,36 @@ def delete(self):
...
 
+meta_schema = Draft4Validator.META_SCHEMA.copy()
+meta_schema['definitions']["simpleTypes"]['enum'].append('int')
+
+default_types = Draft4Validator.DEFAULT_TYPES.copy()
+default_types['int'] = int
+
+
+ExtendedDraft4Validator = \
+    create(meta_schema,
```
Comment (hfaran) [inline tests/func_test.py]: Maybe move the `create(` up and then have the arguments on individual lines. Looks cleaner that way.

File: tornado_json/schema.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
 import json
+
```
Comment (hfaran) [inline tornado_json/schema.py]: I'd prefer to not have these additional newlines (this and the two below it). I don't think they add anything to the readability of the code.

File: tests/func_test.py [hunk 4]
```diff
@@ -1,5 +1,7 @@
-import sys
 import json
+import sys
+
+from jsonschema.validators import Draft4Validator, create
```
Comment (hfaran) [inline tests/func_test.py]: You can get rid of this newline; third-party imports should be in the same section, so it otherwise looks good.

**Comment by hfaran** [comment]
Good stuff! Let me know when you've fixed the few things I've mentioned and then it's good to go.

**Comment by mauler** [comment]
Fixed.

**Pull request merged**